### PR TITLE
fix: pass route extras in routerLink

### PR DESCRIPTION
**Pull request by Akshat55**
Closes carbon-design-system/carbon-components-angular#2904

#### Changelog

**Changed**

* Now calling `navigate` function on anchor link click which will pass in routeExtras.

### CODE REVIEW

File: src/ui-shell/sidenav/side-nav.component.spec.ts [hunk 1]
```diff
@@ -38,7 +38,7 @@ class SideNavTest {
 	}
 }
 
-describe("SideNav", () => {
+fdescribe("SideNav", () => {
```
Comment (Licen-it) [inline src/ui-shell/sidenav/side-nav.component.spec.ts:41]:
```suggestion
describe("SideNav", () => {
```

**Review by Licen-it — CHANGES_REQUESTED** [review]

**Review by Licen-it — APPROVED** [review]

**Comment by klaascuvelier** [comment]
Hey, I am a bit late here with feedback as this MR has been merged already. 

I think the current implementation effectively breaks the idea of the original change. The idea was to rely on the functionality of the directives solely to do navigation and navigation state checks, skipping any custom JS. This would allow for correct cmd + click behavior and so on.
I think the correct fix here would have been to add the navigation extras into the template where now the onclick handler was added.

**Comment by Licen-it** [comment]
@klaascuvelier yeah it would make sense....i didn't think about the cmd + click, but the only pain would be that then passed into the template they wouldn't be a nice since object, but each property would need to be specified separately.
I have an idea @Akshat55, but not sure if that would be ok for carbon team...we could have a directive extending RouterLink directive which is accepting a single object of type NavigationExtras and in the constructor setting each property in this object to the parent class (in a loop way so that if they add/remove any, you won't have to update this code)

**Comment by klaascuvelier** [comment]
Thanks for you reply @Licen-it.
I think adding _all_ the navigation extras in the template of this component should not be too much work, just a one time job.
I'll draft something up quickly asap.

**Comment by Licen-it** [comment]
> Thanks for you reply @Licen-it. I think adding _all_ the navigation extras in the template of this component should not be too much work, just a one time job. I'll draft something up quickly asap. I'll start by creating a new issue so we can take the discussion there. Thanks for collaborating on this and my apologies for breaking stuff :)

Well I was mainly thinking about future proof solution, so that even if a later angular version is changing it will work fine...something to easily support different versions of angular without breaking

**Pull request merged**